### PR TITLE
WIP: Cta trial ended

### PR DESCRIPTION
**Pull request by femacedogomes**

### CODE REVIEW

File: src/app/app/(application)/layout.tsx [hunk 1]
```diff
@@ -2,30 +2,43 @@
...
+import isTrialEndedUser from "@/utils/isTrialEndedUser";
 
 const AppLayout = ({ children }: { children: React.ReactNode }) => {
   const { data: session } = useSession();
   const { isUpdate } = useCompleteSupplierRegister();
-  const [isPremiumExpired, setIsPremiumExpired] = useState(false);
-
+  const [showModal, setShowModal] = useState(false);
   const isCommomUser = useUserRoles().isCommonUser();
+  const sawTrialEndedCTM: boolean = getItemByLocalStorage("sawTrialEndedCTM");
```
Comment (pedrohbl03) [inline src/app/app/(application)/layout.tsx:27]:
```suggestion
  const showTrialEndedCTA: boolean = getItemByLocalStorage("showTrialEndedCTA");
```

File: src/components/mols/TrialConversionModal/index.tsx [hunk 2]
```diff
@@ -11,15 +10,13 @@ import { useSession } from "next-auth/react";
...
+interface ITrialConversionModal {
+  setShowModal: Dispatch<SetStateAction<boolean>>;
   userId: string;
 }
 
-const PremiumConversionModal: React.FC<IPremiumConversionModal> = ({
-  isPremiumExpired,
-  setIsPremiumExpired,
+const TrialConversionModal: React.FC<ITrialConversionModal> = ({
+  setShowModal,
```
Comment (pedrohbl03) [inline src/components/mols/TrialConversionModal/index.tsx:19]: this is a bad patterner -  "prop drilling"

Comment (pedrohbl03) [reply src/components/mols/TrialConversionModal/index.tsx:19]: create a function named "handleModal" and pass this rather than useState

**Review by pedrohbl03 — CHANGES_REQUESTED** [review]

File: src/app/app/(application)/layout.tsx (hunk 1 above)
Comment (femacedogomes) [reply src/app/app/(application)/layout.tsx:27]: feito

File: src/components/mols/TrialConversionModal/index.tsx (hunk 2 above)
Comment (femacedogomes) [reply src/components/mols/TrialConversionModal/index.tsx:19]: done

**Pull request merged**